### ISSUE TITLE
Left bend mistaken for a turn

### ISSUE THREAD

**Issue by polarbearing** [osmandapp/Osmand#3714]
Wrong turn instruction: 
Behind this sharp-angled [junction](https://www.openstreetmap.org/node/255561544), Hohenzollerndamm makes a slight left bend. While routing straight along Hohenzollerndamm, osmand mistakes the bend for a turn, and recommends a regular "left turn into Hohenzollerndamm", and picks the left turn lane. However that would lead into Rudolstädter Str (sharp left). 
Live screenshot below, issue confirmed in simulation mode.
Map: Germany/Brandenburg 2017-April,  osmand 2.7.0#21265 ;
![screenshot_20170409-150645](https://cloud.githubusercontent.com/assets/8266355/25479102/038e45da-2b44-11e7-9b6f-a6b419420111.png)

**Comment by vshcherb** [comment]
The error is slightly different https://www.openstreetmap.org/way/318557887. That road doesn't have any lane dedicate to the right turn, though the right turn is possible as well as left turn. Cause there are 3 possible directions and only 2 kind of lanes, the algorithm wrongly assigned one to another. The issue is pretty difficult to fix without proper data.

**Comment by polarbearing** [comment]
Is that a case there #3638 would help? 
```turn:lanes:evaluated=left|through|through|through|through;right```?

**Comment by vshcherb** [comment]
@polarbearing yes, evaluated would help. Cause it is not clearly drawn here so might be data could be changed in that case to ```turn:lanes=left|through|through|through|through;right```

**Comment by polarbearing** [comment]
Just checked mapillary, there is no ground truth (signposting or so) for tagging the turn:lanes other than 'none'.
So it might be a test case for going forward with turn:lanes:evaluated.

However, my personal pseudocode for the evaluation seems simple enough?
- name lanes A|B|C|D|E from left to right.
- E is 'none', there is a road to the right with one lane, thus we can fill E=through;right
- D is 'none', since E can go through, no right-turn traffic can cross this, thus D=through
- B, C are 'none', the road going straight has 4 lanes in total, thus B=C=through
- E is =left from tagging, and the left road has one lane, match.
